src/azure-cli-core/azure/cli/core/mock.py

```diff
@@ -16,15 +16,19 @@ def __init__(self, commands_loader_cls=None, **kwargs):
         from azure.cli.core.azlogging import AzCliLogging
         from azure.cli.core.cloud import get_active_cloud
         from azure.cli.core.parser import AzCliCommandParser
+        from azure.cli.core.util import random_string
         from azure.cli.core._config import GLOBAL_CONFIG_DIR, ENV_VAR_PREFIX
         from azure.cli.core._help import AzCliHelp
         from azure.cli.core._output import AzOutputProducer
 
         from knack.completion import ARGCOMPLETE_ENV_NAME
 
+        random_config_dir = kwargs.get('random_config_dir', False)
+
         super(DummyCli, self).__init__(
             cli_name='az',
-            config_dir=GLOBAL_CONFIG_DIR,
+            config_dir=os.path.join(GLOBAL_CONFIG_DIR, 'dummy_cli_config_dir',
+                                    random_string()) if random_config_dir else GLOBAL_CONFIG_DIR,
```

Comment:
Well, sorry I didn't make my point clear. I am simply providing an alternative solution. You solution is actually better since your location is easy to find. 😉